presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/GeometryUtils.java

```diff
@@ -152,45 +223,57 @@ public static OGCGeometry deserialize(Slice shape)
         }
         BasicSliceInput input = shape.getInput();
 
-        int spatialReferenceId = input.readInt();
-        SpatialReference spatialReference = null;
-        if (spatialReferenceId != SPATIAL_REFERENCE_UNKNOWN) {
-            spatialReference = SpatialReference.create(spatialReferenceId);
-        }
-
         // GeometryCollection: spatialReferenceId|len-of-shape1|bytes-of-shape1|len-of-shape2|bytes-of-shape2...
         List<OGCGeometry> geometries = new ArrayList<>();
-        while (input.available() > 0) {
-            int length = input.readInt();
-            ByteBuffer buffer = input.readSlice(length).toByteBuffer().slice().order(LITTLE_ENDIAN);
-            Geometry esriGeometry = local().execute(0, Unknown, buffer);
-            OGCGeometry geometry = createFromEsriGeometry(esriGeometry, spatialReference);
-            geometries.add(geometry);
+
+        if (input.available() > 0) {
+            byte code = input.readByte();
+            while (input.available() > 0) {
+                OGCGeometry geometry = extractGeometry(code, input);
+
+                geometries.add(geometry);
+            }
         }
 
         if (geometries.isEmpty()) {
-            return new OGCConcreteGeometryCollection(emptyList(), spatialReference);
+            return new OGCConcreteGeometryCollection(emptyList(), null);
         }
         else if (geometries.size() == 1) {
             return geometries.get(0);
         }
-        return new OGCConcreteGeometryCollection(geometries, spatialReference);
+        return new OGCConcreteGeometryCollection(geometries, null);
+    }
+
+    private static OGCGeometry extractGeometry(byte code, BasicSliceInput input)
+    {
+        GeometryTypeName geometryTypeName = valueOf(code);
+        int length;
+        if (geometryTypeName == GeometryTypeName.GEOMETRY_COLLECTION) {
+            length = input.readInt();
+        }
+        else {
+            length = (int) input.length() - 1;
+        }
+        ByteBuffer buffer = input.readSlice(length).toByteBuffer().slice().order(LITTLE_ENDIAN);
+        Geometry esriGeometry = local().execute(0, Unknown, buffer);
+        return createFromEsriGeometry(esriGeometry, null);
     }
 
     public static Slice serialize(OGCGeometry input)
     {
-        int spatialReferenceId = input.SRID();
         DynamicSliceOutput sliceOutput = new DynamicSliceOutput(100);
-        sliceOutput.appendInt(spatialReferenceId);
 
+        sliceOutput.appendByte(valueOf(input.geometryType()).getCode());
```

Comment:
You can use build-in enum method `ordinal()` and revert changes to GeometryTypeName enum.